tools/crm_attribute.c

```diff
@@ -395,25 +350,27 @@ main(int argc, char **argv)
 
     // Use default CIB location if not given
     if (options.type == NULL) {
-        if (promotion_score) {
+        if (options.promotion_score) {
             // Updating a promotion score node attribute
-            options.type = "reboot";
+            options.type = strdup("reboot");
 
         } else if (options.dest_uname != NULL) {
             // Updating some other node attribute
-            options.type = "forever";
+            options.type = strdup("forever");
```

Comment:
Since we're strdup'ing now, let's just strdup the actual final values rather than strdup reboot and forever just to get freed below (the remapping is still needed in case the user specified reboot or forever)

Reply:
Ah yes, I think I've got it.  That looks much better.